pkgs/by-name/ca/castxml/package.nix

```
@@ -39,32 +39,28 @@ stdenv.mkDerivation (finalAttrs: {
   buildInputs = [
     libffi
     libxml2
+    llvm
     zlib
   ] ++ lib.optionals (!stdenv.isDarwin) [
     libclang
   ];
 
-  cmakeFlags = [
-    (lib.cmakeOptionType "path" "CLANG_RESOURCE_DIR" "${lib.getDev libclang}")
+  cmakeFlags = let clangVersionMajor = builtins.elemAt (lib.strings.splitString "." libclang.version) 0; in [
```

```suggestion
  cmakeFlags = let
    clangVersionMajor = builtins.elemAt (lib.strings.splitString "." libclang.version) 0; 
  in [
```

P.S.: isn't better to use `lib.major`?

https://ryantm.github.io/nixpkgs/functions/library/versions/#function-library-lib.versions.major

---

P.P.S.: there is few to no need of using blocks of indirectio,  when the function is already on `lib` scope.

---

Done.